c_api/IndexScalarQuantizer_c.cpp

```diff
@@ -42,3 +43,80 @@ int faiss_IndexScalarQuantizer_new_with(
     }
     CATCH_AND_HANDLE
 }
+
+DEFINE_DESTRUCTOR(IndexIVFScalarQuantizer)
+DEFINE_INDEX_DOWNCAST(IndexIVFScalarQuantizer)
+
+/// quantizer that maps vectors to inverted lists
+DEFINE_GETTER_PERMISSIVE(IndexIVFScalarQuantizer, FaissIndex*, quantizer)
+
+/// number of possible key values
+DEFINE_GETTER(IndexIVFScalarQuantizer, size_t, nlist)
+/// number of probes at query time
+DEFINE_GETTER(IndexIVFScalarQuantizer, size_t, nprobe)
+DEFINE_SETTER(IndexIVFScalarQuantizer, size_t, nprobe)
+
+/// whether object owns the quantizer
+DEFINE_GETTER(IndexIVFScalarQuantizer, int, own_fields)
+DEFINE_SETTER(IndexIVFScalarQuantizer, int, own_fields)
+
+int faiss_IndexIVFScalarQuantizer_new_with(
+        FaissIndexIVFScalarQuantizer** p_index,
+        FaissIndex* quantizer,
+        size_t d,
+        size_t nlist,
+        FaissQuantizerType qt) {
+    try {
+        auto q = reinterpret_cast<Index*>(quantizer);
+        auto qt_ = static_cast<faiss::ScalarQuantizer::QuantizerType>(qt);
+        IndexIVFScalarQuantizer* index =
+                new IndexIVFScalarQuantizer(q, d, nlist, qt_);
+        *p_index = reinterpret_cast<FaissIndexIVFScalarQuantizer*>(index);
+        return 0;
+    }
+    CATCH_AND_HANDLE
+}
+
+int faiss_IndexIVFScalarQuantizer_new_with_metric(
+        FaissIndexIVFScalarQuantizer** p_index,
+        FaissIndex* quantizer,
+        size_t d,
+        size_t nlist,
+        FaissQuantizerType qt,
+        FaissMetricType metric,
+        int encode_residual) {
+    try {
+        auto q = reinterpret_cast<Index*>(quantizer);
+        auto mt = static_cast<faiss::MetricType>(metric);
+        auto er = (bool)encode_residual;
```

Comment:
Prefer more conservative casts constructs from C++ over C-style casts.
Not that it would be necessary here, as `int` converts implicitly to a `bool` once passed to the constructor.
```suggestion
        auto er = static_cast<bool>(encode_residual);
```